lib/hooks/gitlab/templates/note.html.haml

```diff
@@ -0,0 +1,38 @@
+:ruby
+  note            = payload.object_attributes
+  has_url         = note.url?
+  has_user        = payload.user?
+  repository_name = has_url ? note.url.split('/', 5)[-2] : '' # http://example.com/owner/:name/issues/:id
+  has_issue       = payload.issue?
+  has_merge       = payload.merge_request?
+  has_snippet     = payload.snippet?
+  has_commit      = payload.commit?
+
+%p
+  - if has_user
+    %span= avatar_image_tag payload.user.avatar_url
+    = payload.user.name
+    commented on
+    = note.noteable_type
+  - else
+    The commented on #{note.noteable_type}:
+
+  - if has_issue && has_url
+    %a{href: note.url} #{repository_name}##{payload.issue.iid}
+    %b= payload.issue.title
+    
+  - if has_merge && has_url
+    %a{href: note.url} #{repository_name}##{payload.merge_request.iid}
+    %b= payload.merge_request.title
+  
+  - if has_snippet && has_url
+    %a{href: note.url} #{repository_name}##{payload.snippet.id}
+    %b= payload.snippet.title  
+
+  - if has_commit && has_url
+    %a{href: note.url} #{repository_name}##{payload.commit.id}
+
+  - if !has_url
```

Comment:
`unless has_url` is better (`!` is missable)

Reply:
@kaznum 
fixed.
```
 - unless has_url
    #{repository_name}##{note.id}
```